manylinux1 fails at inference_anakin_api

all manylinux1 fail at inference_anakin_api
![image](https://user-images.githubusercontent.com/6836917/42038098-ad0b40c6-7b1c-11e8-9743-d1dd5d529355.png)
https://paddleci.ngrok.io/viewLog.html?tab=buildLog&buildTypeId=Manylinux1_Cuda90cudnn7avxMkl&buildId=399&_focus=25966
```
[13:37:23]make[3]: *** No rule to make target `../inference_anakin_api', needed by `CMakeFiles/contrib_anakin_inference_lib'.  Stop.
[13:37:23]make[2]: *** [CMakeFiles/contrib_anakin_inference_lib.dir/all] Error 2
[13:37:23]make[2]: *** Waiting for unfinished jobs....
[13:37:23]copying /paddle/paddle/contrib/inference/paddle_inference_api.h -> /paddle/build/fluid_install_dir/contrib/inference
[13:37:23]copying /paddle/build/paddle/contrib/inference/libpaddle_inference_api.* -> /paddle/build/fluid_install_dir/contrib/inference
[13:37:23][ 60%] Built target contrib_inference_lib
[13:37:23][ 60%] Built target parallel_executor
[13:37:23]Scanning dependencies of target paddle_fluid_api
[13:37:23][ 60%] Building CXX object paddle/fluid/inference/CMakeFiles/paddle_fluid_api.dir/io.cc.o
[13:37:27][ 60%] Linking CXX static library libpaddle_fluid_api.a
[13:37:27][ 60%] Built target paddle_fluid_api
[13:37:27]make[1]: *** [CMakeFiles/inference_lib_dist.dir/rule] Error 2
[13:37:27]make: *** [inference_lib_dist] Error 2
```
The reason is
https://github.com/PaddlePaddle/Paddle/blob/fa1fb12b938b5281c1c42b21131658712dff6439/paddle/contrib/inference/CMakeLists.txt#L64-L68
`inference_anakin_api` only compiled in `if (WITH_ANAKIN AND WITH_TESTING)`

but `contrib_anakin_inference_lib` compiled in `if (WITH_ANAKIN)`
https://github.com/PaddlePaddle/Paddle/blob/fa1fb12b938b5281c1c42b21131658712dff6439/cmake/inference_lib.cmake#L157-L158

--- a/paddle/contrib/inference/CMakeLists.txt
+++ b/paddle/contrib/inference/CMakeLists.txt
@@ -61,7 +61,7 @@ cc_library(paddle_inference_tensorrt_subgraph_engine
 inference_api_test(test_paddle_inference_api_tensorrt_subgraph_engine ARGS test_word2vec)
 endif()
 
-if (WITH_ANAKIN AND WITH_TESTING) # only needed in CI
+if (WITH_ANAKIN) # only needed in CI
     # Due to Anakin do not have official library releases and the versions of protobuf and cuda do not match Paddle's,
     # so anakin library will not be merged to our official inference library. To use anakin prediction API, one need to
     # compile the libinference_anakin_api.a and compile with anakin.so.
@@ -71,10 +71,12 @@ if (WITH_ANAKIN AND WITH_TESTING) # only needed in CI
     target_compile_options(inference_anakin_api_shared BEFORE PUBLIC ${ANAKIN_COMPILE_EXTRA_FLAGS})
     target_link_libraries(inference_anakin_api anakin anakin_saber_common)
     target_link_libraries(inference_anakin_api_shared anakin anakin_saber_common)
-    cc_test(inference_anakin_test SRCS paddle_inference_api_anakin_engine_tester.cc
+    if (WITH_TESTING)
+        cc_test(inference_anakin_test SRCS paddle_inference_api_anakin_engine_tester.cc
                                   ARGS --model=${ANAKIN_INSTALL_DIR}/mobilenet_v2.anakin.bin
                                   DEPS inference_anakin_api)
-    target_compile_options(inference_anakin_test BEFORE PUBLIC ${ANAKIN_COMPILE_EXTRA_FLAGS})
+        target_compile_options(inference_anakin_test BEFORE PUBLIC ${ANAKIN_COMPILE_EXTRA_FLAGS})
+     endif(WITH_TESTING)
 endif()
 
 if(WITH_TESTING)

